pkg/netdevice/pciNetDevice_test.go

```diff
@@ -121,6 +124,133 @@ var _ = Describe("PciNetDevice", func() {
+
+			f := &mocks.ResourceFactory{}
+			f.On("GetDefaultInfoProvider", AnythingOfType("string"), "vfio-pci").Return(func(addr string, driver string) types.DeviceInfoProvider {
+				f := factory.NewResourceFactory("fake", "fake", true)
+				return f.GetDefaultInfoProvider(addr, "vfio-pci")
+			}).
+				On("GetRdmaSpec", "0000:00:00.1").Return(rdma1).
+				On("GetRdmaSpec", "0000:00:00.2").Return(rdma2)
+
+			in1 := &ghw.PCIDevice{Address: "0000:00:00.1"}
+			in2 := &ghw.PCIDevice{Address: "0000:00:00.2"}
+
+			It("should populate Rdma device specs if isRdma", func() {
+				defer fs.Use()()
+				defer utils.UseFakeLinks()()
+				dev, err := netdevice.NewPciNetDevice(in1, f, rc)
+
+				Expect(dev.GetDriver()).To(Equal("vfio-pci"))
+				Expect(dev.GetNetName()).To(Equal("eth0"))
+				Expect(dev.GetEnvVal()).To(Equal("0000:00:00.1"))
+				Expect(dev.GetDeviceSpecs()).To(HaveLen(4)) // /dev/vfio/vfio0 and default /dev/vfio/vfio + 2x Rdma devs
+				Expect(dev.GetLinkType()).To(Equal("fakeLinkType"))
+				Expect(dev.GetAPIDevice().Topology.Nodes[0].ID).To(Equal(int64(0)))
+				Expect(dev.GetNumaInfo()).To(Equal("0"))
+				Expect(err).NotTo(HaveOccurred())
+			})
+			It("but not otherwise", func() {
+				defer fs.Use()()
+				defer utils.UseFakeLinks()()
+				dev, err := netdevice.NewPciNetDevice(in2, f, rc)
+
+				Expect(dev.GetDriver()).To(Equal("vfio-pci"))
+				Expect(dev.GetNetName()).To(Equal("eth1"))
+				Expect(dev.GetEnvVal()).To(Equal("0000:00:00.2"))
+				Expect(dev.GetDeviceSpecs()).To(HaveLen(2)) // /dev/vfio/vfio0 and default /dev/vfio/vfio
+				Expect(dev.GetLinkType()).To(Equal("fakeLinkType"))
+				Expect(dev.GetAPIDevice().Topology.Nodes[0].ID).To(Equal(int64(0)))
+				Expect(dev.GetNumaInfo()).To(Equal("0"))
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+		Context("With needsVhostNet", func() {
+			rc := &types.ResourceConfig{
+				ResourceName:   "fake",
+				ResourcePrefix: "fake",
+				SelectorObj: &types.NetDeviceSelectors{
+					NeedVhostNet: true,
+				},
+			}
+
+			fs := &utils.FakeFilesystem{
+				Dirs: []string{
+					"sys/bus/pci/devices/0000:00:00.1/net/eth0",
+					"sys/kernel/iommu_groups/0",
+					"sys/bus/pci/drivers/vfio-pci",
+					"dev/vhost-net",
+				},
+				Symlinks: map[string]string{
+					"sys/bus/pci/devices/0000:00:00.1/iommu_group": "../../../../kernel/iommu_groups/0",
+					"sys/bus/pci/devices/0000:00:00.1/driver":      "../../../../bus/pci/drivers/vfio-pci",
+				},
+				Files: map[string][]byte{"sys/bus/pci/devices/0000:00:00.1/numa_node": []byte("0")},
+			}
+
+			f := factory.NewResourceFactory("fake", "fake", true)
+			in := &ghw.PCIDevice{Address: "0000:00:00.1"}
+			It("should add the vhost-net deviceSpec", func() {
+				defer fs.Use()()
+				defer utils.UseFakeLinks()()
+
+				dev, err := netdevice.NewPciNetDevice(in, f, rc)
+
+				Expect(dev.GetDriver()).To(Equal("vfio-pci"))
+				Expect(dev.GetNetName()).To(Equal("eth0"))
```

Comment:
if driver is vfio-pci we would not have a netdev name right ?

Reply:
Done